src/collect/mod.rs

```diff
@@ -22,19 +22,59 @@ static PERF_DFLT_AUX_BUFSIZE: LazyLock<size_t> = LazyLock::new(|| {
 
 const PERF_DFLT_INITIAL_TRACE_BUFSIZE: size_t = 1024 * 1024; // 1MiB
 
-/// The interface offered by all trace collectors.
-pub trait TraceCollector: Send + Sync {
-    /// Obtain a `ThreadTraceCollector` for the current thread.
-    ///
-    /// A thread may obtain multiple `ThreadTraceCollector`s but must only collect a trace with one
-    /// at a time.
-    ///
-    /// FIXME: This API needs to be fixed:
-    /// https://github.com/ykjit/hwtracer/issues/101
+thread_local! {
+    /// When `Some` holds the `ThreadTraceCollector` that is collecting a trace of the current
+    /// thread.
+    static THREAD_TRACE_COLLECTOR: RefCell<Option<Box<dyn ThreadTraceCollector>>> = RefCell::new(None);
```

Comment:
By the way, I think I can make this `RefCell` a `Cell` later.

Reply:
That's unlikely because `Cell` requires its contents to be `Copy`.

Reply:
Ah, maybe not then.

Reply:
I'll mark this one resolved :)